agent/utils/errors.go

```diff
@@ -18,6 +18,12 @@ import (
 	"strings"
 )
 
+// NamedError defines an interface that wraps error and add additional 'ErrorName' method
+type NamedError interface {
```

Comment:
What do you think about moving this to "api/errors" package? The generic 'utils' package doesn't seem like a good home for this.

Reply:
I've moved this file into api/errors/error_types.go.